src/explore-education-statistics-admin/src/pages/publication/components/__tests__/ReleaseContributorPermissions.test.tsx

```diff
@@ -134,12 +148,61 @@ describe('ReleaseContributorPermissions', () => {
     );
 
     expect(onUserRemove).not.toBeCalled();
+    expect(onUserInvitesRemove).not.toBeCalled();
 
     userEvent.click(within(modal).getByRole('button', { name: 'Confirm' }));
 
     await waitFor(() => {
       expect(onUserRemove).toHaveBeenCalledTimes(1);
       expect(onUserRemove).toHaveBeenCalledWith('user-1');
+      expect(onUserInvitesRemove).not.toBeCalled();
+    });
+  });
+
+  test('cancel invite', async () => {
+    const onUserRemove = jest.fn();
+    const onUserInvitesRemove = jest.fn();
+
+    render(
+      <ReleaseContributorsPermissions
+        contributors={[]}
+        invites={testInvites}
+        onUserRemove={onUserRemove}
+        onUserInvitesRemove={onUserInvitesRemove}
+      />,
+    );
+
+    await waitFor(() => {
+      expect(screen.getByRole('table')).toBeInTheDocument();
```

Comment:
I think we should be using `findByRole` here instead of wrapping this assertion in a `waitFor`. i.e. 
``` 
expect(screen.findByRole('table').toBeInTheDocument()
```

We should avoid using getByRole inside a waitFor as it can cause tests to fail due to getBByRole being slow

https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-waitfor-to-wait-for-elements-that-can-be-queried-with-find

Reply:
In general, I've generally found that the `find*` selectors rarely work properly so my recommendation is to still prefer using `waitFor`.

Also, mentioned this on Slack a while back, but we shouldn't be using `getByRole` (or its variants) within a `waitFor` anyway. It's an expensive assertion which can block the `waitFor` from finishing when used on larger components, potentially causing intermittent test failures.

The recommendation is to use exclusively use more performant selectors `getByText` or `getByTestId` in `waitFor` e.g.:

```ts
await waitFor(() => {
  expect(screen.getByText('something that should appear')).toBeInTheDocument();
});

// Now it's fine to use `getByRole` outside of `waitFor`
expect(screen.geyByRole('table')).toBeInTheDocument()
```

Reply:
Done